node/add-to-path.js

```diff
@@ -14,63 +14,92 @@ const getLinkPath = () => (isMac() || isLinux() ? "/usr/local/bin/oni2" : "")
 
 const isAddedToPath = () => {
     try {
-        fs.lstatSync(getLinkPath());
+        fs.lstatSync(getLinkPath())
     } catch (_) {
         return false
     }
     return true
 }
-const removeFromPath = () => {
+
+const getAppDirectory = () => {
+    if (isMac()) {
+        // Valid for an install from a dmg.
+        return path.join(path.dirname(process.mainModule.filename), "..")
+    } else {
+        // Valid path for an AppImage or Linux tar.gz
+        return path.join(path.dirname(process.mainModule.filename), "..", "..")
+    }
+}
+
+const getOptions = () => {
+    let appDirectory = getAppDirectory()
+
+    // Valid for an install from a dmg.
+    let imgPath = path.join(appDirectory, "Onivim2.icns")
+
+    if (!fs.existsSync(imgPath)) {
+        // Valid path when in a development build.
+        imgPath = path.join(appDirectory, "assets", "images", "Onivim2.icns")
+    }
+
+    if (!fs.existsSync(imgPath)) {
+        // Valid path for an AppImage or Linux tar.gz
+        imgPath = path.join(appDirectory, "..", "Onivim2.icns")
+    }
+
+    const options = { name: "Oni2", icns: imgPath }
```

Comment:
Need to check this...its not showing up in my VM, but that could just be the weird flavour of Ubuntu I'm running...should try in regular ol Ubuntu or something like CentOS or Arch.

Reply:
Looks like this doesn't matter, since I can't get an icon to show up on Ubuntu at all (I can't see anywhere one would go). Its set correctly for the valid locations at least.

Reply:
I don't know what you're doing and I could be completely off, but `.icns` isn't a [supported format in Desktop Entries](https://wiki.archlinux.org/index.php/Desktop_entries#Common_image_formats), so maybe that also applies here.